### ISSUE TITLE
Merriweather demo export not as expected

### ISSUE THREAD

**Issue by davelab6** [mattlag/Glyphr-Studio#176]
https://github.com/davelab6/example-bellweather/pull/1 has a full diff. 

- [x] Question: In `maxp` we see there are 450 glyphs in the original and 326 in the export, but I guess you might have subset it for the demo, as the import has a 'latin only?' feature.

### NAME table Copyright & License Notice

- [x] `name` table ID 0 is the copyright notice, 7 is trademark, 8 and 9 are designer attribution, and 13 and 14 are the licensing information. Its really really important to retain them for copyright, trademark and moral rights laws; currently all 6 are simply dropped. 

- [x] Also, all PlatformID=1 values can be dropped, they are for MacOS 9 and no longer needed at all and just take up space in the file.

### Rename Family

- [x] Merriweather is a trademark of Sorking Type Co, and additionally the font is licensed under the SIL Open Font License with a Reserved Font Name, so for a demo where you play around with modified versions, its really really important to start with a different name, eg `Bellweather`, so users don't inadvertently redistribute modified versions without changing the name.

### Weight Class (quick to fix)

- [ ] `OS/2` table had `<usWeightClass value="400"/>` but it became `<usWeightClass value="500"/>` which is **urgent** because 400 is the 'regular' weight value. 

### Style naming

- [x] File name is `MerriweatherSans-normal.otf`, expecting `MerriweatherSans-Regular.otf` as the style name in the file name.
- [x] Similarly, `CFF ` table in original has `    <CFFFont name="MerriweatherSans-Regular">`, export without edits has `    <CFFFont name="MerriweatherSans-normal">`
- [x] Similarly, in the `name` table, IDs 3, 4, 6, 17 are using `normal` when they should be `Regular`

### Pass ots

The export fonts should pass https://github.com/khaledhosny/ots - without passing ots, https://github.com/mattlag/Glyphr-Studio/issues/62 won't be possible, because the fonts can't be reinserted into the browser for use by its text layout engine. 

```sh
$ ots MerriweatherSans-normal.otf 
cmap: Out of order end range (0 <= 0)
cmap: failed to parse table
Failed to sanitise file!
$
```

### Vertical metrics

- [x] Metrics going wrong somehow: ![screen shot 2015-05-08 at 6 26 59 pm](https://cloud.githubusercontent.com/assets/261579/7541815/dd69047e-f5af-11e4-9db7-a3d6f9d0a2ae.png)

In `OS/2` table the values changed a little, but perhaps this is due to the subsetting of glyphs (so a few of the tallest glyphs are now gone)

```diff
-    <sTypoAscender value="2014"/>		+    <sTypoAscender value="2000"/>
     <sTypoDescender value="-560"/>		     <sTypoDescender value="-560"/>
     <sTypoLineGap value="0"/>		     <sTypoLineGap value="0"/>
-    <usWinAscent value="2014"/>		+    <usWinAscent value="2000"/>
     <usWinDescent value="560"/>		     <usWinDescent value="560"/>
```

and in `hhea` table:

```diff
-    <ascent value="2014"/>		+    <ascent value="2000"/>
```

### Glyph names

- [x] `GlyphOrder` table shows all the glyph names are written using long descriptive names, like `<GlyphID id="94" name="Right Curly Bracket"/>`. For a binary export, this isn't ideal; in a production font file, the glyph names should follow the https://github.com/adobe-type-tools/agl-aglfn specification. 

Additionally, this suggests that the Glyphr API ought to support the 3 kinds of glyph names, `glyph.name_export`, `glyph.name` and `glyph.name_description`, for the glyph names used for export, the glyph names used commonly by designers (eg, following the glyphsapp conventions, eg Џ: `uni040F`, `Dzhe-cy`, `Cyrillic Capital Letter Dzhe`

- [x] Some glyphnames become `<GlyphID id="96" name="[name not found]"/>` which suggests there's a bug in the name parsing

#### Version numbers lost
- [x] `head` table, the `<fontRevision value="1.003"/>` became `<fontRevision value="1.0"/>` but should not change. 
- [x] `name` table ID5 has the same loss

#### Dates lost
- [ ] `head` table, the `<created value="Mon Feb 25 14:05:00 2013"/>` became `<created value="Thu Jan  1 00:00:00 1970"/>` but should not change
- [ ] `head` table `<modified value="Tue Feb 26 07:20:57 2013"/>` is good to change, but instead of last action in History's date (or perhaps local time of export?) it is also `<modified value="Thu Jan  1 00:00:00 1970"/>`

- [ ] This 1970 date is also used for new fonts, but the`created` date is known and displayed in the History panel, as `Initial State`:

![you know the date :)](https://cloud.githubusercontent.com/assets/261579/7548178/67175cc4-f5f7-11e4-80ae-fbe1be003225.png)

#### NAME table font ID

- [ ] The vendor string is lost. I think its okay to remake it - but if we do, instead of
```diff
-      EbenSorkin: MerriweatherSans Regular: 2013		+      :Merriweather Sans normal
```

we should fill in the user and year items, eg,

```diff
-      EbenSorkin: MerriweatherSans Regular: 2013		+      GlyphrUser: Merriweather Sans Regular: 2015
```

#### Edit description?

- [ ] Currently `name` ID 10, the description, is dropped. That isn't ideal, but not too urgent. However, the `input` for editing it should be a `textarea` instead, so you can write more:

![screen shot 2015-05-09 at 1 51 54 am](https://cloud.githubusercontent.com/assets/261579/7547864/196d2a8e-f5ee-11e4-82ec-3b05f40a502a.png)

#### Misc

- [ ] `OS/2` table `<panose>` values were all lost :( Tracked in https://github.com/mattlag/Glyphr-Studio/issues/85

- [ ] `achVendID` was lost, became `XXXX`

- [ ] `post` table lost underline values:

```diff
-    <underlinePosition value="-280"/>		+    <underlinePosition value="0"/>
-    <underlineThickness value="137"/>		+    <underlineThickness value="0"/>
```

- [ ] `GSUB` table was lost, containing **kerning,** but that's a problem of upstream opentype.js

**Comment by mattlag** [comment]
It looks like vertical and horizontal metrics... i'll take a look.

**Comment by davelab6** [comment]
My guess is that the upm was misinterpreted

**Comment by mattlag** [comment]
It looks like the UPM is correct, but some of the other stuff doesn't match up... OpenType.js only has an option for UPM, and individual glyph maxes, so I guess the next step is to make sure those glyph maxes and glyph advanceWidth are correct...

![image](https://cloud.githubusercontent.com/assets/5897669/7544043/74b3e74a-f57e-11e4-9452-1bcb55333ae9.png)
 
```
 <head>
    <!-- Most of this table will be recalculated by the compiler -->
    <tableVersion value="1.0"/>
    <fontRevision value="1.0"/>
    <checkSumAdjustment value="0x24283948"/>
    <magicNumber value="0x5f0f3cf5"/>
    <flags value="00000000 00000000"/>
    <unitsPerEm value="2048"/>
    <created value="Thu Jan 01 00:00:00 1970"/>
    <modified value="Thu Jan 01 00:00:00 1970"/>
    <xMin value="-95"/>
    <yMin value="-560"/>
    <xMax value="2303"/>
    <yMax value="2000"/>
    <macStyle value="00000000 00000000"/>
    <lowestRecPPEM value="0"/>
    <fontDirectionHint value="2"/>
    <indexToLocFormat value="0"/>
    <glyphDataFormat value="0"/>
  </head>

  <hhea>
    <tableVersion value="1.0"/>
    <ascent value="2000"/>
    <descent value="-560"/>
    <lineGap value="0"/>
    <advanceWidthMax value="2479"/>
    <minLeftSideBearing value="0"/>
    <minRightSideBearing value="-228"/>
    <xMaxExtent value="2398"/>
    <caretSlopeRise value="1"/>
    <caretSlopeRun value="0"/>
    <caretOffset value="0"/>
    <reserved0 value="0"/>
    <reserved1 value="0"/>
    <reserved2 value="0"/>
    <reserved3 value="0"/>
    <metricDataFormat value="0"/>
    <numberOfHMetrics value="326"/>
  </hhea>


  <maxp>
    <tableVersion value="0x5000"/>
    <numGlyphs value="326"/>
  </maxp>

  <OS_2>
    <version value="3"/>
    <xAvgCharWidth value="1209"/>
    <usWeightClass value="500"/>
    <usWidthClass value="5"/>
    <fsType value="00000000 00000000"/>
    <ySubscriptXSize value="650"/>
    <ySubscriptYSize value="699"/>
    <ySubscriptXOffset value="0"/>
    <ySubscriptYOffset value="140"/>
    <ySuperscriptXSize value="650"/>
    <ySuperscriptYSize value="699"/>
    <ySuperscriptXOffset value="0"/>
    <ySuperscriptYOffset value="479"/>
    <yStrikeoutSize value="49"/>
    <yStrikeoutPosition value="258"/>
    <sFamilyClass value="0"/>
    <panose>
      <bFamilyType value="0"/>
      <bSerifStyle value="0"/>
      <bWeight value="0"/>
      <bProportion value="0"/>
      <bContrast value="0"/>
      <bStrokeVariation value="0"/>
      <bArmStyle value="0"/>
      <bLetterForm value="0"/>
      <bMidline value="0"/>
      <bXHeight value="0"/>
    </panose>
    <ulUnicodeRange1 value="10000000 00000000 00000000 00001111"/>
    <ulUnicodeRange2 value="00000000 00000000 00000000 00000000"/>
    <ulUnicodeRange3 value="00000000 00000000 00000000 00000000"/>
    <ulUnicodeRange4 value="00000000 00000000 00000000 00000000"/>
    <achVendID value="XXXX"/>
    <fsSelection value="00000000 00000000"/>
    <fsFirstCharIndex value="0"/>
    <fsLastCharIndex value="567"/>
    <sTypoAscender value="2000"/>
    <sTypoDescender value="-560"/>
    <sTypoLineGap value="0"/>
    <usWinAscent value="2000"/>
    <usWinDescent value="560"/>
    <ulCodePageRange1 value="00000000 00000000 00000000 00000000"/>
    <ulCodePageRange2 value="00000000 00000000 00000000 00000000"/>
    <sxHeight value="1140"/>
    <sCapHeight value="1522"/>
    <usDefaultChar value="0"/>
    <usBreakChar value="32"/>
    <usMaxContex value="0"/>
  </OS_2>

```

**Comment by davelab6** [comment]
I have prepared a comparison of the original and the export: 

https://github.com/davelab6/example-bellweather/pull/1

I'll write up tasks in the top of the issue based on this.

**Comment by davelab6** [comment]
Okay, I'm done :)

**Comment by mattlag** [comment]
I'll work on the copyright info stuff right away, I want to make sure that is all in a good state.  Still working on the height metrics stuff...

**Comment by mattlag** [comment]
I included all the import / export options in the (just released) v1.00.03 - all of the specific table stuff is handled by OpenType.js.  Unfortunately, OpenType.js does not allow for direct writing ability to tables, and only exposes the following top level font metadata:

 - copyright
 - description
 - designer
 - designerURL
 - familyName
 - license
 - licenseURL
 - manufacturer
 - manufacturerURL
 - styleName
 - trademark
 - version

but for all of these I import and export accordingly.

**Comment by davelab6** [comment]
it sucks that ttx won't run in the browser, its API is really nice as you can work at whatever level you want :)

**Comment by davelab6** [comment]
I'll retest this at the weekend or so

**Comment by mattlag** [comment]
The fix for this was to set the UPM in OpenType.js to 1000, even if the UPM was greater than that - i'm going to file a bug with OpenType.js on this, it doesn't seem right, and the highlighted area for the characters seems off.

fixed in v1.00.03
